exp/internal/controllers/machinepool_controller_test.go

```diff
@@ -302,6 +305,7 @@ func TestReconcileMachinePoolRequest(t *testing.T) {
 	testCases := []struct {
 		machinePool expv1.MachinePool
 		expected    expected
```

Comment:
Let's improve these tests:
* let's add a name field and describe what these test cases are doing and use that name field in l.511 with t.Run instead of the MachinePool names
* let's please add verification that the MachinePools are actually gone or not. If I see correctly we currently only check for errors

Reply:
done

Reply:
Thx!